src/managers/config.py

```diff
@@ -460,8 +480,34 @@ def controller_listener(self) -> None:
         pass  # TODO: No good abstraction in place for the controller use case
 
     @property
-    def client_listeners(self) -> list[Listener]:
+    def extra_listeners(self) -> list[Listener]:
         """Return a list of extra listeners."""
+        extra_host_baseports = [
+            tuple(listener.split(":")) for listener in self.config.extra_listeners
+        ]
+
+        extra_listeners = []
+        extra_count = 0
+        for host, baseport in extra_host_baseports:
+            for auth_map in self.state.enabled_auth:
+                host = host.replace("{unit}", str(self.state.unit_broker.unit_id))
+                extra_listeners.append(
+                    Listener(
+                        host=host,
+                        auth_map=auth_map,
+                        scope="EXTRA",
+                        baseport=int(baseport),
+                        extra_count=extra_count,
```

Comment:
This class is getting pretty big. Maybe time to extract some of the functionality into separate classes for different configuration tasks?

Reply:
For now I think we'll keep it as it would be out of scope here.
But you're right, if users are looking for greater customisation of this, it would be a good time to revisit how we're building the listeners generally.